### ISSUE TITLE
Improve error messages for base functions

### ISSUE THREAD

**Issue by dbp** [logiccomp/lsl#19]
Rosette has pretty bad messages for at least some functions; e.g., if i pass the wrong value to `max`:

```
(max #t #f)
```

I get the following error:

```
. . [assert] curried:extreme: expected real? arguments
  arguments: (#t)
```

Maybe we can wrap (reasonably named) contracts around them in the process of re-exporting them?

**Comment by camoy** [comment]
Yes, exporting these with the right contract should improve the error message.

**Comment by dbp** [comment]
No more Rosette; closing :)
